lib/spack/env/cc

```diff
@@ -98,25 +98,25 @@ case "$command" in
     cpp)
         mode=cpp
         ;;
-    cc|c89|c99|gcc|clang|icc|pgcc|xlc)
+    cc|c89|c99|gcc|clang|icc|pgcc|xlc|xlc_r)
```

Comment:
I run on the current develop branch and use:
```yaml
- compiler:
    environment: {}
    extra_rpaths: []
    flags:
      cflags:   -axCORE-AVX2,AVX
      cxxflags: -axCORE-AVX2,AVX
      fflags:   -axCORE-AVX2,AVX
    modules: []
    operating_system: centos6
    paths:
      cc:  /soft/intel/15.0.1/composer_xe_2015.0.090/bin/intel64/icc
      cxx: /soft/intel/15.0.1/composer_xe_2015.0.090/bin/intel64/icpc
      f77: /soft/intel/15.0.1/composer_xe_2015.0.090/bin/intel64/ifort
      fc:  /soft/intel/15.0.1/composer_xe_2015.0.090/bin/intel64/ifort
    spec: intel@15.0.0
    target: x86_64
```
So aside from adding a `xl_r` compiler, you probably shouldn't have to modify anything to pass in the flags you want.